section3/reflection.md

```diff
@@ -2,12 +2,24 @@
 
 1. In Section 3, we removed even more supports and pushed you to learn a bit more independently, read through technical documentation, and use your resources more. How did that feel?
 
+- It felt challenging but I was able to complete the section. I like to explore and understand the contents, so was pretty smooth.
+
 2. Think of a web or mobile application you use regularly. Where is an example of a place that the developers probably use Arrays? Explain.
 
+- The tags on Instagram. You place a tag on the search bar and will search in all the user's posts for posts that contain that tag.
+
 3. In your own words, explain what an Array method is.
 
+- Array is a variable that can store a list of values under a single variable name.
+
 4. What is the purpose of a loop? When we would use a loop in conjunction with array vs. without?
 
+- A loop help us to execute a block of codes for a number of times. We use a loop in conjunction with an array when we want to execute a specific action in all the elements from the array, and we use a loop without an array when we need to repeat a string/number until a condition is met.
+
 5. If you had to teach a 5 year old about loops, how would you explain it?
 
+- A loop is an action that check a condition and then run a code block. The loop will continue to check and run until the condition is false.
+
 6. What questions do you still have about Arrays and loops?
+
+- I understand the theory for loops and arrays, but I don't really know where I would apply those. I know that we are starting with the basics, but not being able to se it in the real world makes me very lost.
```

Comment:
Great question!

So one example might be - in my app I have a database of users emails. I want to send out a promotional email to all my users. Instead of writing the code that triggers the email for each individual user, I can write it once, inside a loop that iterates over each email address, and every user gets that email.

Another could connect to your Instagram tag idea. If a user types "crossfit" into the search bar, code will run - it will loop though all posts, and inside that loop, use a conditional that essentially checks "does any info in this post contain the string of crossfit?" - the algorithm is surely much more complicated than that, but at some level, they have to search through all the posts (which requires looping through those arrays of data).